### DISCUSSION TITLE
BMAX B2 Plus does not work with M.2 PCIe Coral EdgeTPU

### DISCUSSION THREAD

**Discussion by shred86** [blakeblackshear/frigate#1901] [General]
I ordered the BMAX B2 Plus as it was one of the mini PCs listed as recommended hardware but unfortunately, it appears the M.2 PCIe Coral EdgeTPU does not work with this device. It's not detected at all by the system but when an M.2 NVMe is used, it's detected just fine.

I actually also have an AWOW AK41 mini PC I ordered which is almost identical to the BMAX B2 Plus, but it has the same issue where the M.2 PCIe Coral EdgeTPU is not detected. Both the AWOW AK41 and BMAX B2 Plus run the same BIOS build version V1.01_101(06/24/2021).

I've only found one low powered mini PC that seems to work with the M.2 PCIe Coral EdgeTPU, the AWOW AL34, but I've been running into random freezing issues with it hence the reason I'm looking for an alternative.

Is anyone out there running a BMAX B2 Plus or any low power mini PC with an M.2 PCIe Coral EdgeTPU?

Edit: This is what `lspci` displays:

````
00:00.0 Host bridge: Intel Corporation Gemini Lake Host Bridge (rev 06)
00:00.1 Signal processing controller: Intel Corporation Celeron/Pentium Silver Processor Dynamic Platform and Thermal Framework Processor Participant (rev 06)
00:00.3 System peripheral: Intel Corporation Celeron/Pentium Silver Processor Gaussian Mixture Model (rev 06)
00:02.0 VGA compatible controller: Intel Corporation GeminiLake [UHD Graphics 600] (rev 06)
00:0e.0 Multimedia audio controller: Intel Corporation Celeron/Pentium Silver Processor High Definition Audio (rev 06)
00:0f.0 Communication controller: Intel Corporation Celeron/Pentium Silver Processor Trusted Execution Engine Interface (rev 06)
00:12.0 SATA controller: Intel Corporation Celeron/Pentium Silver Processor SATA Controller (rev 06)
00:13.0 PCI bridge: Intel Corporation Gemini Lake PCI Express Root Port (rev f6)
00:13.2 PCI bridge: Intel Corporation Gemini Lake PCI Express Root Port (rev f6)
00:13.3 PCI bridge: Intel Corporation Gemini Lake PCI Express Root Port (rev f6)
00:14.0 PCI bridge: Intel Corporation Gemini Lake PCI Express Root Port (rev f6)
00:15.0 USB controller: Intel Corporation Celeron/Pentium Silver Processor USB 3.0 xHCI Controller (rev 06)
00:16.0 Signal processing controller: Intel Corporation Celeron/Pentium Silver Processor Serial IO I2C Host Controller (rev 06)
00:16.1 Signal processing controller: Intel Corporation Celeron/Pentium Silver Processor Serial IO I2C Host Controller (rev 06)
00:17.0 Signal processing controller: Intel Corporation Device 31b4 (rev 06)
00:1c.0 SD Host controller: Intel Corporation Celeron/Pentium Silver Processor SDA Standard Compliant SD Host Controller (rev 06)
00:1e.0 SD Host controller: Intel Corporation Device 31d0 (rev 06)
00:1f.0 ISA bridge: Intel Corporation Celeron/Pentium Silver Processor LPC Controller (rev 06)
00:1f.1 SMBus: Intel Corporation Celeron/Pentium Silver Processor Gaussian Mixture Model (rev 06)
02:00.0 Ethernet controller: Realtek Semiconductor Co., Ltd. RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller (rev 10)
03:00.0 Network controller: Realtek Semiconductor Co., Ltd. RTL8821CE 802.11ac PCIe Wireless Network Adapter
````

**Comment by blakeblackshear** [comment]
I use the USB Coral with the BMAX B2. I haven't had much luck finding mini pc's with support for the m.2 coral. This one seems promising: 

Mini PC, Odyssey X86 Blue J4125 Windows 10 Desktop Computer, Intel Celeron Processor 8GB RAM and 128GB SSD with Re_Computer Case, Dual Gigabit Ethernet, Dual Band Wi-Fi/Bluetooth5.0, 4K HD,Dual Screen https://www.amazon.com/dp/B08FD3QHQC/ref=cm_sw_r_apan_glt_fabc_9NAWZ0S4YJFPJW9J9F86

**Reply by YANMRXYB** [reply]
Can anyone share the copy file from BMAX B2 PLUS BIOS?

**Comment by shred86** [comment]
Ah, that looks nearly perfect for what I need. Thanks for the link, just ordered one. ;)

Yeah, I’ve gone through three Mini PCs now. The AWOW AL34 was the only one that worked but the random freezing after ~3-7 days makes it unreliable. I’ve found a couple other Mini PCs with M.2 PCIe slots but they’re sometimes located beneath the M.2 slot intended for an SSD which wouldn’t work well with something like the Coral Edge TPU (too much heat being immediately next to an SSD).

**Reply by patdemko** [reply]
I also just got a BMAX B2 Plus based on that recommendations page and tried using it with a m.2 dual edge TPU using [this E-key to M-key](https://www.amazon.com/gp/product/B079NB8J3B/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1) adapter but I also found it didn't work. I also have a USB coral and that works fine with the BMAX B2 Plus so I've been using that.  BTW, I was able to get the m.2 dual edge TPU setup I described above to work in an older NUC I have so I'd imagine it'd probably work on a newer one too.

@blakeblackshear maybe it'd be worth noting somewhere on your hardware recommendations page which accelerators have been confirmed to work with which mini PC setups to help anyone out looking to get one?  Although it may need to be crowd sourced information though since not sure how many different accelerator types or mini pc's you have to test with.

**Comment by shred86** [comment]
Just received the Odyssey X86 Blue (Intel Celeron J4125) today and set it up. Works perfectly with the M.2 PCIe Coral Edge TPU. This might be the perfect low energy mini server for me. Running ten Docker containers (Frigate, Home Assistant, Pihole, Unifi Controller, etc.) with 4x 720p camera streams into Frigate and only sitting around 22% CPU usage. Thermals are significantly better with this device than other Mini PCs I tested as well. Based on some YouTube reviews I saw, it was measured at about 4W idle and 15W under full load. Very nice little device. Thanks @blakeblackshear for the point out!

**Reply by blakeblackshear** [reply]
Awesome. I will add it to the hardware page.

**Reply by blakeblackshear** [reply]
@shred86 are you using the M.2 B+M version?

**Reply by shred86** [reply]
Yep, M.2 B+M.

**Reply by shred86** [reply]
Came home today to find everything frozen. Rebooted the device and found a bunch of errors in the system logs that looked memory related. Sure enough, ran memtest86+ again and it's failing big time. The strange part is the first thing I did when I received it yesterday was run memtest86+ and it passed just fine... less than 24 hours and the memory has already gone bad. :(

**Reply by blakeblackshear** [reply]
Bummer

**Reply by shred86** [reply]
Well, new one came in yesterday so everything is back up and running. So far so good. I’ll probably run memtest86+ again in a few weeks just to make sure it’s still good (passed when I tested it yesterday).

**Reply by salvobellino95** [reply]
Does it have working hw acceleration?

**Reply by shred86** [reply]
Yeah, it has the Intel J4125 which supports hw accel.

**Reply by salvobellino95** [reply]
@shred86  Does you recommend this mini pc?

**Reply by shred86** [reply]
~~I didn’t keep it because it did not work with my M.2 PCIe Coral EdgeTPU. I think other folks are using it with a USB Coral EdgeTPU.~~

My mistake. Thought you were asking about the BMax. The Odyssey X86 does work with the M.2 PCIe Coral EdgeTPU.

**Reply by salvobellino95** [reply]
So what are you using now? I have to buy another device because I have a RockPi4c+ which is Rockchip-based and I haven't found a working way to enable hwaccel. In the Coral website the M.2 is recommended for production, so I think maybe it is faster than the USB version?

**Reply by shred86** [reply]
Ah sorry I thought you were asking about the BMax. I am using the Odyssey X86. It does work with the M.2 Coral EdgeTPU. 

I don’t think there’s really a noticeable difference between the USB and M.2 Coral EdgeTPU.

**Comment by kabakakao** [comment]
anyone out there having success with the coral on the bmax b2 plus?
It looks really nice for that price

**Comment by chy520cvv** [comment]
A year later my BMAX B2 Plus J4115 is stuck on the logo screen. Can't enter BIOS with Delete, ESC, F7, F2.  
Support on website is minimal, no answer to email.
Anyone will help me?

**Reply by takkischitt** [reply]
Did you ever find a solution for this?
